test/client/components/CommunityJoinLinkHandler.test.js

```diff
@@ -0,0 +1,61 @@
+require('../support')
+import { mocks, helpers } from '../../support'
+import CommunityJoinLinkHandler from '../../../src/containers/community/CommunityJoinLinkHandler.js'
+import { renderIntoDocument } from 'react-addons-test-utils'
+import { NAVIGATE } from '../../../src/actions'
+const { createElement } = helpers
+
+describe('CommunityJoinLinkHandler', () => {
+  var store, redirectUrl
+
+  const setup = props => {
+    const component = createElement(CommunityJoinLinkHandler, props, {store})
+    renderIntoDocument(component).getWrappedInstance()
+  }
+
+  beforeEach(() => {
+    store = mocks.redux.store({
+      people: {current: {id: 42}},
+      errors: {}
+    })
+    store.transformAction(NAVIGATE, action => redirectUrl = action.payload)
+  })
+
+  it('redirects to onboarding with new membership', done => {
+    store.getState().people.current.memberships = [{
+      community: {id: '1', name: 'Foomunity', slug: 'foomunity'}
+    }]
+    setup({params: {id: 'foomunity', tagName: 'bar'}})
+    // this is to compensate for the setTimeout in CommunityJoinLinkHandler
+    setTimeout(() => {
```

Comment:
I'm not in a promise chain here, so can't I just use setTimeout in the
definition of renderAndWait?

On Wed, Jun 22, 2016 at 1:11 PM, Lawrence Wang <notifications@github.com>
wrote:

> In test/client/components/CommunityJoinLinkHandler.test.js
> <https://github.com/Hylozoic/hylo-redux/pull/65#discussion_r68125671>:
>
> > +
> > +  beforeEach(() => {
> > +    store = mocks.redux.store({
> > +      people: {current: {id: 42}},
> > +      errors: {}
> > +    })
> > +    store.transformAction(NAVIGATE, action => redirectUrl = action.payload)
> > +  })
> > +
> > +  it('redirects to onboarding with new membership', done => {
> > +    store.getState().people.current.memberships = [{
> > +      community: {id: '1', name: 'Foomunity', slug: 'foomunity'}
> > +    }]
> > +    setup({params: {id: 'foomunity', tagName: 'bar'}})
> > +    // this is to compensate for the setTimeout in CommunityJoinLinkHandler
> > +    setTimeout(() => {
>
> look at helpers.wait in test/support, which is a promisified version of
> setTimeout -- using it, you can integrate the setTimeout(fn, 10) bit that's
> repeated in all of these tests into setup, e.g.:
>
> // renamed setup to renderAndWaitrenderAndWait({params: {id: 'foomunity', tagName: 'bar'}}, () => {
>   expect(redirectUrl).to.equal('/c/foomunity/onboarding')
> })
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Hylozoic/hylo-redux/pull/65/files/b9fcf7b01a39ec5ba6bb4a1670073302b3cf0042#r68125671>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA2Y9KlBhtE2oYtbhKIP2igCJP1Oivs9ks5qOZbVgaJpZM4I4y8H>
> .
>



-- 
Robbie Carlton

719.358.3482
<robbie@authenticmanprogram.com>